QuickFIXn/SocketReader.cs

```diff
@@ -207,6 +213,15 @@ protected void ProcessStream()
 
         protected void DisconnectClient()
         {
+            _readCancellationTokenSource?.Cancel();
+            _readCancellationTokenSource?.Dispose();
+            _readCancellationTokenSource = null;
+
+            // just wait when read task will be cancelled
```

Comment:
So the unobserved `SocketException`s were coming from closing the stream when disconnecting, and now we are cancelling the read with the `CancellationTokenSource`, and the `ContinueWith` is there to swallow the resulting `OperationCanceledException` on the `Task`. Is that right? Perhaps it is worth expanding the comment?

```suggestion
            // The ContinueWith observes (and swallows) the resulting
            // OperationCanceledException on the task.
            // This avoids unobserved exceptions appearing elsewhere
            // (which we ultimately don't care about during disconnect)
```

Reply:
Hey @Rob-Hague . I am sorry for the delayed response. Not exactly.  The unobserved exception was triggered when we call `.BeginRead(...)` and the other side just closed the connection.   I reimplemented the socket reading based on Begin/End async approach to the task-based approach that will raise the exception on task awaits. 
About the `ContinueWith ` you are absolutely right.